website/common/script/ops/buy/abstractBuyOperation.js

```diff
@@ -139,6 +139,10 @@ export class AbstractGoldItemOperation extends AbstractBuyOperation {
     if (item && item.canOwn && !item.canOwn(user)) {
       throw new NotAuthorized(this.i18n('cannotBuyItem'));
     }
+
+    if (item.locked) {
+      throw new NotAuthorized(this.i18n('cannotBuyItem'));
+    }
```

Comment:
I think this is causing the other test failure. Note that above we short-circuit if `item` isn't available--apparently we use this code for a subscriber to buy Gems using Gold, but there isn't a full `item` to represent a Gem.

So if we were doing this, the fix would be to change the condition line to `if (item && item.locked) {`
...but, I think we backed off making changes to `abstractBuyOperation` anyway? I don't think these conditions are ever going to hit, given the buy operation doesn't have access to `locked`. I think the actual correction to be made for this PR is to remove the changes to this file.

Reply:
Took it out!

Reply:
looks like it's still there?

Reply:
I'm not seeing it? Here's the current code for `canUserPurchase` on the branch, lines 131 through 150:

```
  canUserPurchase (user, item) {
    this.item = item;
    const itemValue = this.getItemValue(item);

    const userGold = user.stats.gp;

    if (userGold < itemValue * this.quantity) {
      throw new NotAuthorized(this.i18n('messageNotEnoughGold'));
    }

    if (item && item.canOwn && !item.canOwn(user)) {
      throw new NotAuthorized(this.i18n('cannotBuyItem'));
    }
  }

  async subtractCurrency (user, item) {
    const itemValue = this.getItemValue(item);

    user.stats.gp -= itemValue * this.quantity;
  }
```